stdlib/public/Darwin/Accelerate/vDSP_Arithmetic.swift

```diff
@@ -0,0 +1,3303 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2019 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+// Vector-vector and vector-scalar arithmetic
+
+extension vDSP {
+    
+    // MARK: c[i] = a[i] + b                                vDSP_vsadd
+    
+    /// Returns the elementwise sum of `vector` and `scalar`,
+    /// single-precision.
+    ///
+    /// - Parameter scalar: the `b` in `c[i] = a[i] + b`.
+    /// - Parameter vector: the `a` in `c[i] = a[i] + b`.
+    /// - Returns: The `c` in `c[i] = a[i] + b`.
+    @inline(__always)
+    @available(iOS 9999, macOS 9999, tvOS 9999, watchOS 9999, *)
+    public static func add<U>(_ scalar: Float,
```

Comment:
This is a design point, and I know that this is an Apple framework and not necessarily subject to SE or any community involvement, but I'll posit that lots of developers who _could_ use these functions _won't_ use them because they're hard to discover. `vDSP.add(6, myArray)` is certainly nicer than the C name `vDSP_vsadd` but still, if you didn't already know it exists, you'd never find it.

Did you consider putting these functions on `AccelerateBuffer` directly? Alternatively, namespacing them would also be more discoverable than hanging them off of `vDSP` as static functions, IMO:

```swift
extension AccelerateBuffer {
  func vectorised() -> vDSP<Self> { return vDSP(data: self) }
}

struct vDSP<T: AccelerateBuffer> {
  var data: T
}
extension vDSP where T.Element == Double {
  func add(scalar: T.Element) {}
}

[1.0, 2, 3].vectorised().add(scalar: 6)
```

Just a thought.

Reply:
@karwa We definitely wouldn't want them on `AB` directly because we don't want every [Float] to pop up ~500 autocomplete suggestions unrelated to what most people are doing (worse, in some editors, the autocomplete suggestions don't take into account conditional conformance, so they'd appear on every Array; that's an editor bug, but we don't need to compound the pain).

Your vDSP wrapping function is plausible, but it's also worth considering that most of these operations do not have a primary source that makes a good LHS. Ultimately, this design was up to the Accelerate team, but I think they struck a reasonable compromise.